src/Persistence/Array_.php

```diff
@@ -328,72 +328,65 @@ protected function setLimitOrder($m, &$action)
+            // condition must have 3 params at these point
+            if (count($cond) != 3) {
+                // condition can have up to three params
                 throw new Exception([
-                    'Condition not acceptable for Array persistence',
-                    'condition' => $cond,
+                    'Persistence\Array_ driver condition unsupported format',
+                    'reason' => 'condition can have two to three params',
+                    'condition'=> $cond,
                 ]);
-                /*
-                // OR conditions
-                if (is_array($cond[0])) {
-                    foreach ($cond[0] as &$row) {
-                        if (is_string($row[0])) {
-                            $row[0] = $m->getElement($row[0]);
-                        }
-                    }
-                }
-
-                $q->where($cond[0]);
-                continue;
-                */
             }
 
-            if (is_string($cond[0])) {
-                $cond[0] = $m->getElement($cond[0]);
-            }
+            // action
+            $method = strtolower($cond[1]);
 
-            if (count($cond) == 2) {
-                if ($cond[0] instanceof Field) {
-                    $cond[1] = $this->typecastSaveField($cond[0], $cond[1]);
-                }
-                $q->where(is_string($cond[0]) ? $cond[0] : $cond[0]->short_name, $cond[1]);
-            } elseif(count($cond) == 3) {
-                if ($cond[0] instanceof Field) {
-                    $cond[1] = $this->typecastSaveField($cond[0], $cond[1]);
-                }
-                if (!method_exists($q,strtolower($cond[1]))) {
-                    throw new Exception([
-                        'Method ' . $cond[1] . ' not supported by Array persistence driver',
-                        'condition'=> $cond,
-                    ]);
-                }
-                $q->{strtolower($cond[1])}(is_string($cond[0]) ? $cond[0] : $cond[0]->short_name, $cond[2]);
-            } else {
+            // check if the method is supported by the iterator
+            if (!method_exists( $iterator, $method)) {
                 throw new Exception([
-                    'Condition not acceptable for Array persistence',
+                    'Persistence\Array_ driver condition unsupported method',
+                    'reason' => "method $method not implemented for Action\Iterator",
                     'condition'=> $cond,
                 ]);
+            }
 
-                /*
-                if ($cond[0] instanceof Field) {
-                    $cond[2] = $this->typecastSaveField($cond[0], $cond[2]);
-                }
-                $q->where($cond[0], $cond[1], $cond[2]);
-                */
+            // get the model field
+            if (is_string($cond[0])) {
+                $cond[0] = $model->getElement($cond[0]);
+            }
+
+            if (!is_a($cond[0],\atk4\data\Field::class)) {
```

Comment:
we always use `instanceof` instead of `is_a`, not sure why. Maybe because of this https://bugs.php.net/bug.php?id=55475